agents/agents-utils/src/commonMain/kotlin/ai/koog/agents/utils/KtorHttpClient.kt

```diff
@@ -0,0 +1,135 @@
+package ai.koog.agents.utils
+
+import io.github.oshai.kotlinlogging.KLogger
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.call.body
+import io.ktor.client.plugins.sse.SSEClientException
+import io.ktor.client.plugins.sse.sse
+import io.ktor.client.request.accept
+import io.ktor.client.request.headers
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.bodyAsText
+import io.ktor.client.statement.readRawBytes
+import io.ktor.http.ContentType
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpMethod
+import io.ktor.http.isSuccess
+import io.ktor.util.reflect.TypeInfo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.withContext
+import kotlin.reflect.KClass
+
+/**
+ * KtorHttpClient is an implementation of the KoogHttpClient interface, utilizing Ktor's HttpClient
+ * to perform HTTP operations, including POST requests and Server-Sent Events (SSE) streaming.
+ *
+ * This client provides enhanced logging, flexible request and response handling, and supports
+ * configurability for underlying Ktor HttpClient instances.
+ *
+ * @property clientName The name of the client, used for logging and traceability.
+ * @property logger A logging instance of type KLogger for recording client-related events and errors.
+ * @constructor Creates a KtorHttpClient instance with an optional base Ktor HttpClient and configuration block.
+ *
+ * @param baseClient The base Ktor HttpClient instance to be used. Default is a newly created instance.
+ * @param configurer A lambda function to configure the base Ktor HttpClient instance.
+ * The configuration is applied using the Ktor `HttpClient.config` method.
+ */
+public class KtorHttpClient(
```

Comment:
Such name might be confusing, because it's a wrapper around HttpClient from Ktor. What if someone actually would use `KtorHttpClient` thinking it's from Ktor?

Maybe let's make it internal, and add a public extension to `KoogHttpClient`, so that one would use `KoogHttpClient.fromKtor(...)` (or `fromKtorClient`)  -- and the next one would be `KoogHttpClient.fromSpring()`(or `fromSpringClient`), etc.